test_capitals.py

```diff
@@ -0,0 +1,5 @@
+from capitals import count_letters
+
+assert count_letters('nfWEHI') == (4, 2), 'should be 4 uppercase letter and 2 lower case'
```

Comment:
assert scrie ca e un instrument pentru testare si verifica conditia care urmeaza. daca totul e True, nu se intampla nimic, daca e False apare o eroare si pot sa presupun ca apare si textul care l-ai pus intre ghilimele.

capitals e un modul? sau e denumirea fisierului din care se importa functia count_letters pe care tot tu ai definit-o in fisierul cela?

Reply:
cum se porneste test_capitals.py? trebuie sa fie un alt fisier care uneste ambele capitals.py si test_capitals.py?

Reply:
yup, assert aruncă excepție când nu se îndeplinește condiția.

capitals în cazul dat e fișier, python tratează ca modul.

test-capitals.py pornești manual, adica $python test_capitals.py. Dar sunt instrumente de automatizare de gen pytest. Acolo și testele se scriu altfel, nu cu aserturi, și se pornesc ele toate automat, după numele test_*, și raportul e arătat, nu doar excepție când nu e corect.